### PR TITLE
fix small current internal resistance issue

### PR DESCRIPTION
**Pull request by TomTranter**

### CODE REVIEW

File: examples/drive_cycle_comparison.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import pybamm
```
Comment (rtimms) [inline examples/drive_cycle_comparison.py:1]: would be good to add a few more comments in this example explaining that you are comparing solving a model in liiopack and in pybamm

File: liionpack/protocols.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import numpy as np
 
 
-def generate_protocol_from_experiment(experiment):
+def generate_protocol_from_experiment(experiment, flatten=True):
```
Comment (rtimms) [inline liionpack/protocols.py:7]: docstring needs updating

File: liionpack/solvers.py [hunk 3]
```diff
@@ -131,7 +138,8 @@ def solve(
         self.split_models(Nspm, nproc)
 
         # Generate the protocol from the supplied experiment
-        protocol = lp.generate_protocol_from_experiment(experiment)
+        protocol = lp.generate_protocol_from_experiment(experiment, flatten=True)
+        protocol = protocol + [protocol[-1]]
```
Comment (rtimms) [inline liionpack/solvers.py:142]: I'm not sure I follow why you are doing this?

File: liionpack/solvers.py [hunk 4]
```diff
@@ -177,54 +184,68 @@ def solve(
...
-                break
-
-            self.timestep += 1
+        self.fixed_Ri = np.ones(Nspm) * np.nan
+        with tqdm(total=Nsteps, desc="Stepping simulation") as pbar:
+            step = 0
+            while step < Nsteps:
+                # Solver specific steps
+                # self.resting = protocol[step] == 0.0
+                self.resting = (
```
Comment (rtimms) [inline liionpack/solvers.py:193]: what are these used for?

Comment (rtimms) [reply liionpack/solvers.py:193]: oh wait I see

File: liionpack/solvers.py [hunk 5]
```diff
@@ -177,54 +184,68 @@ def solve(
...
+                self.restarting = (
+                    step > 0 and protocol[step] != 0.0 and protocol[step - 1] == 0.0
+                )
+                self.step_actors()
+                self.get_actor_output(step)
+
+                # # Calculate internal resistance and update netlist
+                temp_v = self.output[0, step, :]
+                temp_ocv = self.output[1, step, :]
+                if not self.resting and not self.restarting:
```
Comment (rtimms) [inline liionpack/solvers.py:205]: Could do with a comment explaining why you don't always update temp_Ri

**Review by rtimms — APPROVED** [review]
thanks @TomTranter looks good

File: liionpack/solvers.py (hunk 5 above)
Comment (TomTranter) [reply liionpack/solvers.py:215]: Added

**Pull request merged**